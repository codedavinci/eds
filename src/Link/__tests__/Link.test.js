import React from 'react'
import { Link } from '../'

describe('Link', () => {
  const doShallow = (props = {}) => shallow(<Link {...props} />)

  it('should export', () => {
    expect(typeof Link).toBeTruthy()
  })

  it('renders', () => {
    const link = doShallow()

    expect(link).toMatchSnapshot()
  })

  it('does other things', () => {
    const link = doShallow()

    expect(link).toExist()
  })

  it('passes additional attributes to the element', () => {
    const link = doShallow({ id: 'the-id', 'data-some-attr': 'some value' })

    expect(link).toHaveProp('id', 'the-id')
    expect(link).toHaveProp('data-some-attr', 'some value')
  })

  it('does not allow custom CSS', () => {
    const link = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(link).not.toHaveProp('className', 'my-custom-class')
    expect(link).not.toHaveProp('style')
  })
})
