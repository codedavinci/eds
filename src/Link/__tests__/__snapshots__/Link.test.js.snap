// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `
<styled.button>
  Hello
</styled.button>
`;
